File: Data Structures & Algorithms/median-of-two-sorted-arrays/submission-0.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        i, j, m, n, ans= 0, 0, len(nums1), len(nums2), []
        while(i<m and j<n):
            if(nums1[i]<=nums2[j]):
                ans.append(nums1[i])
                i+=1
            elif nums2[j]<nums1[i]:
                ans.append(nums2[j])
                j+=1
        while(i<m):
            ans.append(nums1[i])
            i+=1
        while(j<n):
            ans.append(nums2[j])
            j+=1
        if len(ans)%2 == 0:
            return (ans[len(ans)//2]+ ans[(len(ans)//2)-1])/2
        else:
            return ans[len(ans)//2]

File: Data Structures & Algorithms/combination-target-sum/submission-0.cpp
class Solution {
public:
    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        vector<int>subsets;
        vector<vector<int>>ans;
        dfs(nums, 0, target, subsets, ans);
        return ans;
    }
    void dfs(vector<int>nums, int i, int target, vector<int>subsets, vector<vector<int>>&ans){
        if(i<nums.size() && target==0) {
            ans.push_back(subsets);
            return;
        }
        if(i==nums.size()||target<0) return;
        subsets.push_back(nums[i]);
        dfs(nums, i, target-nums[i], subsets, ans);
        subsets.pop_back();
        dfs(nums, i+1, target, subsets, ans);
    }
};